notebooks/240-dolly-2-instruction-following/240-dolly-2-instruction-following.ipynb

```diff
@@ -255,6 +242,101 @@
     "    ov_model.save_pretrained(model_path)"
    ]
   },
+  {
+   "cell_type": "markdown",
+   "id": "5b1238c8-dcc9-4495-aeff-1ecbd8bd5082",
+   "metadata": {},
+   "source": [
+    "### NNCF model weights compression [$\\Uparrow$](#Table-of-content:)\n",
+    "\n",
+    "NNCF [Weights Compression algorithm](https://github.com/openvinotoolkit/nncf/blob/develop/docs/compression_algorithms/CompressWeights.md) compresses  weights of a model to `INT8`. This is an alternative to [Quantization algorithm](https://github.com/openvinotoolkit/nncf/blob/develop/docs/compression_algorithms/post_training/Quantization.md) that compresses both weights and activations. Weight compression is effective in optimizing footprint and performance of large models where the size of weights is significantly larger than the size of activations, for example, in Large Language Models (LLMs) such as Dolly 2.0. Additionaly, Weight Compression usually leads to almost no accuracy drop."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "id": "8e5c9e68-3772-432f-b231-f1163442357d",
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "application/vnd.jupyter.widget-view+json": {
+       "model_id": "6be9ab974c06454e81077fe735e7cb37",
+       "version_major": 2,
+       "version_minor": 0
+      },
+      "text/plain": [
+       "Dropdown(description='Compression:', index=1, options=('Disable', 'Enable'), value='Enable')"
+      ]
+     },
+     "execution_count": 5,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "to_compress = widgets.Dropdown(\n",
+    "    options=['Disable', 'Enable'],\n",
+    "    value='Enable',\n",
+    "    description='Compression:',\n",
+    "    disabled=False,\n",
+    ")\n",
+    "to_compress"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "id": "392940e3-01da-4876-a9d1-2475ed3da882",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "* Original IR model size: 10590.42 MB\n",
+      "* Compressed IR model size: 2660.28 MB\n",
+      "* Model compression rate: 3.981\n"
+     ]
+    },
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "Compiling the model...\n",
+      "Set CACHE_DIR to dolly-v2-3b_compressed/model_cache\n"
+     ]
+    }
+   ],
+   "source": [
+    "import nncf\n",
+    "import shutil\n",
+    "import openvino.runtime as ov\n",
+    "\n",
+    "compressed_model_path = Path(f'{model_path}_compressed') / 'openvino_model.xml'\n",
+    "\n",
+    "def compress_model(model):\n",
```

Comment:
I thought we planned to use optimum-intel to compress the model.

Reply:
I haven't found any API for weight compression in optimum-intel, only quantization. @AlexKoff88 is there such API?

Reply:
@nikita-savelyevv , it's the same API. You can find the example here https://github.com/openvinotoolkit/openvino_notebooks/blob/main/notebooks/254-llm-chatbot/254-llm-chatbot.ipynb

Reply:
Waiting for https://github.com/huggingface/optimum-intel/pull/415 to be merged

Reply:
The PR has been merged so you can use the functionality you were waiting for.

Reply:
Blocked now by CVS-121154